view_scene.py

```diff
@@ -0,0 +1,347 @@
+        # TODO (in future PR): use the real image height and width
+        px = K[0, 2]
+        py = K[1, 2]
+        img_w = px * 2
+        img_h = py * 2
+        frustum_obj = ViewFrustum(fx, img_w, img_h)
+
+        edges_worldfr = frustum_obj.get_mesh_edges_worldframe(wTi)
+        for verts_worldfr in edges_worldfr:
+
+            lines = [[0, 1]]
+            # color is in range [0,1]
+            color = tuple(colormap[i].tolist())
+            colors = [color for i in range(len(lines))]
+
+            line_set = open3d.geometry.LineSet(
+                points=open3d.utility.Vector3dVector(verts_worldfr),
+                lines=open3d.utility.Vector2iVector(lines),
+            )
+            line_set.colors = open3d.utility.Vector3dVector(colors)
+            line_sets.append(line_set)
+
+    return line_sets
+
+
+def draw_cameras_mayavi(
+    zcwTw: Pose3, fig: mayavi.core.scene.Scene, calibrations: List[Cal3Bundler], wTi_list: List[Pose3]
+) -> None:
+    """Render camera frustums as collections of line segments, using Mayavi mlab.
+
+    Args:
+        zcwTw: transforms world points to a new world frame where the point cloud is zero-centered
+        fig: Mayavi mlab figure object.
+        calibrations: calibration object for each camera
+        wTi_list: list of camera poses for each image
+    """
+    colormap = np.array(
+        [[color_obj.rgb] for color_obj in Color("red").range_to(Color("green"), len(wTi_list))]
+    ).squeeze()
+
+    for i, (K, wTi) in enumerate(zip(calibrations, wTi_list)):
+        wTi = zcwTw.compose(wTi)
+
+        color = tuple(colormap[i].tolist())
+
+        K = K.K()
+        fx = K[0, 0]
+
+        # Use 2*principal point as proxy measure for image height and width
+        # TODO (in future PR): use the real image height and width
+        px = K[0, 2]
+        py = K[1, 2]
+
+        img_w = px * 2
+        img_h = py * 2
+        frustum_obj = ViewFrustum(fx, img_w, img_h)
+
+        edges_worldfr = frustum_obj.get_mesh_edges_worldframe(wTi)
+        for edge_worldfr in edges_worldfr:
+
+            # start and end vertices
+            vs = edge_worldfr[0]
+            ve = edge_worldfr[1]
+
+            # TODO: consider adding line_width
+            mlab.plot3d(  # type: ignore
+                [vs[0], ve[0]],
+                [vs[1], ve[1]],
+                [vs[2], ve[2]],
+                color=color,
+                tube_radius=None,
+                figure=fig,
+            )
+
+
+def compute_point_cloud_center_robust(point_cloud: np.ndarray) -> np.ndarray:
+    """Robustly estimate the point cloud center.
+
+    Args:
+        point_cloud: array of shape (N,3) representing 3d points.
+
+    Returns:
+        mean_pt: coordinates of central point, ignoring outliers.
+    """
+    ranges = np.linalg.norm(point_cloud, axis=1)
+    outlier_thresh = np.percentile(ranges, 75)
+    mean_pt = point_cloud[ranges < outlier_thresh].mean(axis=0)
+    return mean_pt
```

Comment:
let me know if you think there is a better way to do this. Perhaps largest CC of density-based clustering?

Reply:
clustering sounds like unnecessary computation for visualization. I think this is fine. Alternatively, you could filter out points beyond 2 x sigma or 3 x sigma.